Index: chrome/renderer/render_widget.cc
===================================================================
--- chrome/renderer/render_widget.cc	(revision 26900)
+++ chrome/renderer/render_widget.cc	(working copy)
@@ -424,6 +424,11 @@
     NOTREACHED();
     return;
   }
+  if (!background_.empty() && !background_.isOpaque()) {
+    // If the background contains any transparency, we must clear the output
+    // buffer, or we will end up with garbage from the last render.
+    canvas->drawARGB(0,0,0,0,SkXfermode::kClear_Mode);
+  }
 
   // We may get back a smaller canvas than we asked for.
   damaged_rect.set_width(canvas->getDevice()->width());
@@ -506,6 +511,11 @@
     NOTREACHED();
     return;
   }
+  if (!background_.empty() && !background_.isOpaque()) {
+    // If the background contains any transparency, we must clear the output
+    // buffer, or we will end up with garbage from the last render.
+    canvas->drawARGB(0,0,0,0,SkXfermode::kClear_Mode);
+  }
 
   // We may get back a smaller canvas than we asked for.
   damaged_rect.set_width(canvas->getDevice()->width());
